tests/utils/mock_thread_pool.h

```diff
@@ -42,13 +42,29 @@ class tieredIndexMock {
         size_t k;    // The number of results to return.
         size_t n;    // The number of vectors in the index (might be useful for the mock)
         size_t dim;  // The dimension of the vectors in the index (might be useful for the mock)
-        std::atomic_int &successful_searches; // A reference to a shared counter that counts the
+        std::atomic_int *successful_searches; // A reference to a shared counter that counts the
                                               // number of successful searches.
+        size_t ef;
+        size_t iter;                         // For benchmarks, the number of iteration
+        VecSimQueryResult_List *all_results; // For benchmarks, an array to store the results in
+
+        // Note that some members are not relevant for certain use-cases of the SearchJobMock,
+        // so we use default values that indicates that the member is in use only if an actual
+        // value is sent to the contractor (for example, we use dim in some unit tests, but in
+        // benchmarks this is irrelevant).
         SearchJobMock(std::shared_ptr<VecSimAllocator> allocator, JobCallback searchCB,
-                      VecSimIndex *index_, void *query_, size_t k_, size_t n_, size_t dim_,
-                      std::atomic_int &successful_searches_)
+                      VecSimIndex *index_, size_t k_, void *query_, size_t n_, size_t dim_,
+                      std::atomic_int *successful_searches_)
             : AsyncJob(allocator, HNSW_SEARCH_JOB, searchCB, index_), query(query_), k(k_), n(n_),
-              dim(dim_), successful_searches(successful_searches_) {}
+              dim(dim_), successful_searches(successful_searches_), ef(-1), iter(-1),
+              all_results(nullptr) {}
+        SearchJobMock(std::shared_ptr<VecSimAllocator> allocator, JobCallback searchCB,
+                      VecSimIndex *index_, size_t k_, size_t ef_, size_t iter_,
+                      VecSimQueryResult_List *all_results_)
+            : AsyncJob(allocator, HNSW_SEARCH_JOB, searchCB, index_), query(nullptr), k(k_), n(-1),
+              dim(-1), successful_searches(nullptr), ef(ef_), iter(iter_),
+              all_results(all_results_) {}
+
```

Comment:
a comment about where do we use each? 
also, setting dim and n to -1 seems kinda risky to me

Reply:
Added comments.
Why risky? it's for having an indication that this is an illegal usage and crash/fails ASAP (it's better than having garbage there in my opinion). Also, recall that this is a non-production code...